zea/scan.py

```
@@ -523,6 +538,15 @@ def initial_times(self):
 
         return value[self.selected_transmits]
 
+    @property
+    def t_peak(self):
+        """The time of the peak of the pulse in seconds of shape (n_tw,)."""
+        t_peak = self._params.get("t_peak")
+        if t_peak is None:
+            t_peak = np.array([1 / self.center_frequency])
```

is `1/center_frequency` a kind of default value for the time to peak? If so, why is the default from `compute_time_to_peak` set to `0.0`?

---

`1/center_frequency` is the period of the carrier, so I think it is a reasonable guess for the time to peak. The reasoning to have 0.0 as default argument in `compute_time_to_peak` that the peak of a pulse of `N` samples is always between 0.0 and `N/sampling_frequency`, therefore the peak of a signal of 0 samples should be zero. However, now that I think about it, it might be better to just raise an error as this is clearly a degenerate case. What do you think?